modules/ROOT/pages/mule-maven-plugin.adoc

```diff
@@ -56,730 +39,581 @@ The repository for the plugin is at https://repository.mulesoft.org/nexus/conten
-    </configuration>
-    <executions>
-        <execution>
-            <id>deploy</id>
-            <phase>deploy</phase>
-            <goals>
-                <goal>deploy</goal>
-            </goals>
-        </execution>
-    </executions>
-</plugin>
-----
+To successfully deploy your Mule app to CloudHub, consider the following requirements:
 
-== Deploying to a Mule Server Using the Agent
+* The host and port number of your HTTP Listener flow sources are properly configured. +
++
+If you are using the HTTP Listener as sources for your flow, you need to set its host to *0.0.0.0* and its port to *${http.port}*.
++
+CloudHub then dynamically allocates a port at deployment time.
 
-You can also configure the plugin to deploy to an existing Mule server using the API provided by the Mule agent. In the code shown below, the `uri` parameter is the endpoint of the REST API of the agent.
 
+=== Deploying to CloudHub
+
+. Make sure you added the Mule Maven Plugin to your pom.xml file.
+. inside the plugin element, add a configuration for your CloudHub deployment as shown below:
++
 [source,xml,linenums]
 ----
 <plugin>
-    <groupId>org.mule.tools.maven</groupId>
-    <artifactId>mule-maven-plugin</artifactId>
-    <version>2.2.1</version>
+  ...
     <configuration>
-        <deploymentType>agent</deploymentType>
-        <uri>http://localhost:9999/</uri>
+        <cloudHubDeployment>
+            <uri>https://anypoint.mulesoft.com</uri>
+            <muleVersion>${mule.version}</muleVersion>
+            <username>${username}</username>
+            <password>${password}</password>
+            <applicationName>${cloudhub.application.name}</applicationName>
+            <environment>${environment}</environment>
+            <properties>
+                <key>value</key>
+            </properties>
+        </cloudHubDeployment>
     </configuration>
-    <executions>
-        <execution>
-            <id>deploy</id>
-            <phase>deploy</phase>
-            <goals>
-                <goal>deploy</goal>
-            </goals>
-        </execution>
-    </executions>
+
 </plugin>
 ----
-
-Then run:
-
++
+The example is using placeholder values. Configure each value with your CloudHub information.
+. From the command line in your project's folder, package the app and run the deploy goal:
++
 [source,console,linenums]
 ----
-$ mvn org.mule.tools.maven:mule-maven-plugin:deploy
+> mvn clean package deploy -DmuleDeploy
 ----
 
-For a list and description of the parameters employed, see <<Agent>>.
-
-== Running Integration Tests
+=== Redeploying to CloudHub
 
-One of the most important uses for the plugin is to run integration tests on your integration application. Check the working example in `src/it/standalone/example-integration-tests`.
+To redeploy the app, run the same command as you did to deploy. +
+CloudHub rewrites the app you had deployed.
 
-To run integration tests, the basic steps are the following:
+=== Encrypting credentials
```

Comment:
Could you move this section at the bottom of this doc? Because the server parameter to pass encrypted credentials is available on all the platform deployments (Cloudhub, ARM, Runtime Fabric)